deltachat-ffi/src/lib.rs

```diff
@@ -1390,7 +1390,13 @@ pub unsafe extern "C" fn dc_get_msg_html(
     }
     let ctx = &*context;
 
-    block_on(MsgId::new(msg_id).get_html(&ctx)).strdup()
+    block_on(async move {
+        MsgId::new(msg_id)
+            .get_html(&ctx)
+            .await
+            .map(|s| s.strdup())
+            .unwrap_or_else(ptr::null_mut)
```

Comment:
Can't you just call `...await.strdup()`? This should automatically return null if it's None.

Reply:
you're right, good catch! i'll do a commit.

/me just copy+pasted from another ffi-implementation

Reply:
i think, the whole change is not needed, `block_on(MsgId::new(msg_id).get_html(&ctx)).strdup()` should just do fine :)